src/App/DocumentObjectPyImp.cpp

```diff
@@ -52,11 +52,11 @@ std::string DocumentObjectPy::representation() const
 Py::String DocumentObjectPy::getName() const
 {
     DocumentObject* object = this->getDocumentObjectPtr();
-    const char* internal = object->getNameInDocument();
-    if (!internal) {
+    std::string internal = object->getNameInDocument();

```

Comment:
Should be const reference.

Reply:
This should actually be split... no `internal` is needed.

We should use `isAttachedToDocument()` and if so, simply return `{getNameInDocument()}`.

This is actually how I am doing on the "split PRs":
https://github.com/FreeCAD/FreeCAD/commit/b5160f0cf916c51499a2446a8c6471feb71638bd#diff-14f0ebc3222eb16cf3e7cd368140b2fdd3ae6ad97801c29318cfeb28bc590002R55